### DISCUSSION TITLE
Can't load a package onto my device

### DISCUSSION THREAD

**Discussion by colebugley** [AxisCommunications/acap-native-sdk-examples#65] [Q&A] [answered]
I've tried going through the hello world example presented here. I've gotten to the last step where you load the package onto the unit, but when I try to upload the package it says "this application package is not compatible with this product". Has anyone else encountered this issue or could provide some insight to help me with?

**Comment by pataxis** [comment]
Hi,
it sounds like you have built the hello world application for a different architecture than the one your Axis product has.
My guess is that the application is build with the default `armv7hf` while your Axis product has architecture `aarch64`.

- Check which [architecture](https://axiscommunications.github.io/acap-documentation/docs/get-started/set-up-device.html#check-device-properties) your Axis product has.
- The architecture of the built ACAP application (.eap-file) should be seen in the file name, containing either `armv7hf` or `aarch64`.

To build the example with a different architecture, the hello-world build [instructions](https://github.com/AxisCommunications/acap-native-sdk-examples/tree/master/hello-world#build-the-application) points out a simple build argument to achieve this:
```sh
docker build --build-arg ARCH=aarch64 --tag my-hello-world:1.0 .
```

If this doesn't help, which Axis product are you trying to install the ACAP application to and could you provide the last 20-30 lines of the system log?

**Reply by colebugley** [reply]
Hello,
Thanks for your reply.

I tried building the example with aarch64 as the architecture and received the same error message. The product I'm using is the AXIS F34 Main Unit. 
I'm actually sure which architecture my device uses because when I try to follow the instructions for finding which architecture my device is using it gives me a 404 error and I'm not sure why that is. Every other web page path in the hello world example works for me. I get the same error when I attempt to get the information through the cmd prompt way as well.

Here is the system from my last attempt when attempting to upload the file through the AXIS device manager.
![Capture](https://user-images.githubusercontent.com/55918689/181254039-26f20b1f-d2b0-4d0b-8f3a-8639f693d42d.PNG)
.

Thanks.

**Comment by pataxis** [comment, marked as answer]
Ahh, I see. The F34 is an Artpec-5 product having a mips chip and latest FW is 6.50 from what I can tell from our internal systems. I'm quite sure that this product has reached End of Upgrade and both the [ACAP Native SDK](https://axiscommunications.github.io/acap-documentation/docs/axis-devices.html#acap-native-sdk) and [ACAP 3 SDK](https://help.axis.com/acap-3-developer-guide#compatibility) require newer SoCs and FW.

I'm not sure of the level of support for this product, I will have to wait for some people to get back from vacation. If you are in a partner program you can also try to ask a question to the Axis support.
